pkgs/by-name/we/wemeet/package.nix

```diff
@@ -0,0 +1,305 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  dpkg,
+  autoPatchelfHook,
+  makeWrapper,
+  nss,
+  xorg,
+  desktop-file-utils,
+  libpulseaudio,
+  libgcrypt,
+  dbus,
+  systemd,
+  udev,
+  libGL,
+  fontconfig,
+  freetype,
+  openssl,
+  wayland,
+  libdrm,
+  harfbuzz,
+  openldap,
+  curl,
+  nghttp2,
+  libunwind,
+  alsa-lib,
+  libidn2,
+  rtmpdump,
+  libpsl,
+  libkrb5,
+  xkeyboard_config,
+  libsForQt5,
+  gcc,
+  pkg-config,
+  libgcc,
+  libglvnd,
+  zlib,
+  systemdLibs,
+  nspr,
+  expat,
+  glib,
+  fetchFromGitHub,
+  cmake,
+  wireplumber,
+  libportal,
+  xdg-desktop-portal,
+  opencv,
+  pipewire,
+  fetchgit,
+}:
+let
+  wemeet-wayland-screenshare = stdenv.mkDerivation {
+    pname = "wemeet-wayland-screenshare";
+    version = "0-unstable-2024-11-30";
+
+    src = fetchFromGitHub {
+      owner = "xuwd1";
+      repo = "wemeet-wayland-screenshare";
+      rev = "4d9875270e111f27c5c02e68e9188fe5a4616756";
+      hash = "sha256-on1MFTFQMzMoK6uwjRs3DGnyHWV8ozfATxvlwM4bhZY=";
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [
+      cmake
+      pkg-config
+    ];
+
+    buildInputs = [
+      wireplumber
+      libportal
+      xdg-desktop-portal
+      libsForQt5.qt5.qtwayland
+      libsForQt5.xwaylandvideobridge
+      opencv
+      pipewire
+      xorg.libXdamage
+      xorg.libXrandr
+      xorg.libX11
+    ];
+
+    cmakeFlags = [ "-DCMAKE_BUILD_TYPE=Release" ];
+
+    dontWrapQtApps = true;
+
+    installPhase = ''
+      runHook preInstall
+
+      install -Dm755 ./libhook.so $out/lib/libhook.so
```

Comment:
I encountered an issue where the SSO login is unable to invoke the browser properly.

```
/nix/store/73imdq2ph2rlc5vi4pkbyhf07wy451vr-wemeet-3.19.2.400/bin/wemeetapp 
wemeet:WemeetSatrt
Failed to create variant from json2024-12-10 08:14:03|123224827637888|xnn.XNNHandleMonitor|xnnhandlemonitor.cpp:35||regist handle,type:0 handle:226780032
2024-12-10 08:14:03|123224827637888|xnn.XNNRTResource|xnnrtresource.cpp:20||the acc thread cnt is 0
2024-12-10 08:14:03|123224827637888|xnn.XNNContext|xnncontext.cpp:99||init xnn context succ
2024-12-10 08:14:03|123224827637888|XNNReg|xnnlayerreg.cpp:866||layer map inited
2024-12-10 08:14:03|123224827637888|xnn.XNNHandleMonitor|xnnhandlemonitor.cpp:35||regist handle,type:0 handle:226781024
2024-12-10 08:14:03|123224827637888|xnn.XNNRTResource|xnnrtresource.cpp:20||the acc thread cnt is 0
2024-12-10 08:14:03|123224827637888|xnn.XNNContext|xnncontext.cpp:99||init xnn context succ
2024-12-10 08:14:03|123224827637888|XNNReg|xnnlayerreg.cpp:866||layer map inited
2024-12-10 08:14:03|123224827637888|xnn.XNNHandleMonitor|xnnhandlemonitor.cpp:35||regist handle,type:0 handle:224773584
2024-12-10 08:14:03|123224827637888|xnn.XNNRTResource|xnnrtresource.cpp:20||the acc thread cnt is 0
2024-12-10 08:14:03|123224827637888|xnn.XNNContext|xnncontext.cpp:99||init xnn context succ
sh: symbol lookup error: /nix/store/yihijvy80b6mj8yw5j517kffrnpzyqjk-wemeet-wayland-screenshare-0-unstable-2024-11-30/lib/libhook.so: undefined symbol: XRRGetOutputInfo
sh: symbol lookup error: /nix/store/yihijvy80b6mj8yw5j517kffrnpzyqjk-wemeet-wayland-screenshare-0-unstable-2024-11-30/lib/libhook.so: undefined symbol: XRRGetOutputInfo
```

However, it seems that this is likely an upstream issue.

Reply:
This does not affect normal users, but impacts those using enterprise accounts.

Reply:
You are right, it works well after add `xorg.libXrandr` to `LD_LIBRARY_PATH`.